biz.aQute.repository/src/aQute/bnd/repository/p2/provider/P2Indexer.java

```diff
@@ -130,12 +132,29 @@ private Repository readRepository() throws Exception {
 		P2Impl p2 = new P2Impl(client, this.url, Processor.getExecutor());
 		List<Artifact> artifacts = p2.getArtifacts();
 		List<Promise<Resource>> fetched = new ArrayList<>(artifacts.size());
-		Set<URI> visited = new HashSet<>(artifacts.size());
+		Set<BundleId> visited = new HashSet<>(artifacts.size());
+
+		final Map<BundleId,Resource> existing = new HashMap<>();
+		if (getBridge() != null) {
```

Comment:
To validate it is unchanged, you need to compare the new hash to the old hash. If they are the same, then you can reuse the Resource object. Otherwise you need to process the artifact again.